iree/compiler/Conversion/Common/VectorTransferOptimization.cpp

```diff
@@ -57,6 +57,31 @@ static void eraseDeadAllocAndStores(FuncOp funcOp) {
 
 namespace {
 
+// Pattern to canonialize tranpose where only one dimension is not unit
+// dimension. In this case the transpose is a no-op and should be simplified
+// before getting to the conversion to llvm/spirv.
+// TODO(thomasraoux): This should be moved in
+// `populateCastAwayVectorLeadingOneDimPatterns` but might need more discussion
+// on the semantic of transpose in this case.
+class TransposeUnitDimToShapeCast
```

Comment:
Not sure exactly what this pattern is doing. I dont see any unit tests for this either.

Reply:
I added a unit test.

Reply:
That is a very specific pattern. I see why you need it though. Shouldnt this just live in core.